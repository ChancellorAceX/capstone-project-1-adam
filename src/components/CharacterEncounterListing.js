import React from 'react';
import { Link } from 'react-router-dom';
import DeleteButton from './DeleteButton';

export default class CharacterEncounterListing extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      editing: false
    };
  }
  toggleEdit = (e) => this.setState({ editing: !this.state.editing });
  saveChanges;

  render() {
    const { pcid, pcname, initiative, ac, hp, hpmax, pcbestiaryid } = this.props.character;
    return !this.state.editing ? !!pcbestiaryid ?
      (
        <tr>
          <td className='viewLink'><Link className='viewLink' to={`/bestiary/${pcbestiaryid}`}><button className='viewLink'>View</button></Link></td>
          <td className='name'>{pcname}</td>
          <td className='init'>{initiative}</td>
          <td className='ac'>{ac}</td>
          <td className='hp'>{`${hp}/${hpmax}`}</td>
          <td className='buttons'><button className='editButton' onClick={this.toggleEdit} type='button'>Edit</button>
          <DeleteButton
            target='character'
            id={this.props.character.pcid}
            encounterData={this.props.encounterData}
            setTargetState={this.props.setTargetState}
          /></td>
        </tr >
      )
      :
      (
        <tr>
          <td className='viewLink'></td>
          <td className='name'>{pcname}</td>
          <td className='init'>{initiative}</td>
          <td className='ac'>{ac}</td>
          <td className='hp'>{`${hp}/${hpmax}`}</td>
          <td className='buttons'><button className='editButton' onClick={this.toggleEdit} type='button'>Edit</button>
          <DeleteButton
            target='character'
            id={this.props.character.pcid}
            encounterData={this.props.encounterData}
            setTargetState={this.props.setTargetState}
          /></td>
        </tr >
      )
      :
      (
        <tr>
          <td className='viewLink'></td>
          <td id='editname' className='pcname'>
            <textarea className={`pcname ${pcid}`} type='text' defaultValue={pcname} onChange={this.props.setUpdateObject} />
          </td>
          <td id='editinit' className='initiative'>
            <input className={`initiative ${pcid}`} type='number' defaultValue={initiative} onChange={this.props.setUpdateObject} />
          </td>
          <td id='editac' className='ac'>
            <input className={`ac ${pcid}`} type='number' defaultValue={ac} onChange={this.props.setUpdateObject} />
          </td>
          <td id='edithp' className='hp'>
            <input className={`hp ${pcid}`} type='number' defaultValue={hp} onChange={this.props.setUpdateObject} />
            /
            <input className={`hpmax ${pcid}`} type='number' defaultValue={hpmax} onChange={this.props.setUpdateObject} />
          </td>
          <td id='editbuttons' className='buttons'>
            <button className='saveEdit' onClick={(e) => { this.toggleEdit(e); this.props.updateCharacter(e); }} type='button'>Save
            </button>
          </td>
        </tr>
      );
  }
}