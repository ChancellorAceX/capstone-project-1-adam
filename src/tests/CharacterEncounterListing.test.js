import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CharacterEncounterListing from '../components/CharacterEncounterListing';

describe('CharacterEncounterListing component', () => {
  const props = {
    "character": {
      "pcid": 16,
      "pcname": "Aarakocra1",
      "pcclass": null,
      "level": null,
      "initiative": 5,
      "ac": 12,
      "hp": 24,
      "hpmax": 24,
      "pcbestiaryid": 1,
      "npc": true,
      "pccampaignid": null,
      "pcencounterid": 1
    },
    "updateCharacter": "ƒ () {}",
    "setUpdateObject": "ƒ () {}",
    "encounterData": [
      "{ac: 50, hp: 72, hpmax: 100, initiative: 30, level:…}",
      "{ac: 12, hp: 24, hpmax: 24, initiative: 5, level: n…}",
      "{ac: 12, hp: 15, hpmax: 20, initiative: null, level…}",
      "{ac: 18, hp: 32, hpmax: 32, initiative: null, level…}"
    ],
    "setTargetState": "ƒ () {}",
    "index": 1
  };

  it('renders the full listing component', () => {
    const wrapper = shallow(<CharacterEncounterListing {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});