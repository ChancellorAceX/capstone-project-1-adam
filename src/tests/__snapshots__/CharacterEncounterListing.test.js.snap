// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterEncounterListing component renders the full listing component 1`] = `
<tr>
  <td
    className="viewLink"
  >
    <Link
      className="viewLink"
      to="/bestiary/1"
    >
      <button
        className="viewLink"
      >
        View
      </button>
    </Link>
  </td>
  <td
    className="name"
  >
    Aarakocra1
  </td>
  <td
    className="init"
  >
    5
  </td>
  <td
    className="ac"
  >
    12
  </td>
  <td
    className="hp"
  >
    24/24
  </td>
  <td
    className="buttons"
  >
    <button
      className="editButton"
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
    <DeleteButton
      encounterData={
        Array [
          "{ac: 50, hp: 72, hpmax: 100, initiative: 30, level:…}",
          "{ac: 12, hp: 24, hpmax: 24, initiative: 5, level: n…}",
          "{ac: 12, hp: 15, hpmax: 20, initiative: null, level…}",
          "{ac: 18, hp: 32, hpmax: 32, initiative: null, level…}",
        ]
      }
      id={16}
      setTargetState="ƒ () {}"
      target="character"
    />
  </td>
</tr>
`;
